app/code/Magento/Catalog/Model/Product/Option/Type/Date.php

```diff
@@ -302,7 +305,10 @@ public function prepareOptionValueForRequest($optionValue)
         $infoBuyRequest = $confItem->getOptionByCode('info_buyRequest');
         try {
             $value = $this->serializer->unserialize($infoBuyRequest->getValue());
-            if (is_array($value) && isset($value['options']) && isset($value['options'][$this->getOption()->getId()])
+
+            if (is_array($value)
+                && isset($value['options'])
+                && isset($value['options'][$this->getOption()->getId()])
             ) {
```

Comment:
Hello @karyna-tsymbal-atwix 
Please fix this condition like below

```suggestion
            if (is_array($value) && isset($value['options'][$this->getOption()->getId()])) {
```

Reply:
hi @andrewbess , done